CMakeLists.txt

```diff
@@ -231,55 +231,42 @@ if(WIN32)
 endif()
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
 
-# Get the current working branch
 execute_process(
-  COMMAND git rev-parse --abbrev-ref HEAD
+  COMMAND git describe --tags --always --dirty=-modified
```

Comment:
unfortunately this does not work when at the tag commit. 
In this case only the tag is returned.

Reply:
Add `--long` to always include the commit SHA1.

Reply:
That is a feature, not a bug.